doc/designs/evp_skey.md

```diff
@@ -0,0 +1,183 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represented as octet buffer. It
+may be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
```

Comment:
Replace: This design proposes the use of an opaque type to represent symmetric keys.